File: python/sort01.py
# sort01
def sort(num):
    start = 0; end = len(num) - 1
    while start < end:
        if num[start] == 0:
            start+=1
        elif num[end] == 1:
            end-=1
        else:
            num[start], num[end] = num[end], num[start]
            start += 1
            end -= 1
    print(num)            

num = [1, 1, 0, 1, 1, 1, 0]
sort(num)

File: javaPractice/ARRAY/Sort_0_1.java
package javaPractice.ARRAY;

public class Sort_0_1 {
    public static void main(String[] args) {
        int[] arr = { 1, 0, 0, 1, 0, 1, 0 };
        sort(arr);
    }

    static void sort(int[] arr){
        int s=0, e=arr.length-1;
        while(s<=e){
            if(arr[s]==0)
                s++;
            else if(arr[e]==1)
                e--;
            else if(arr[s]==1 && arr[e]==0 && s<e){
                arr[s++] = 0;
                arr[e--] = 1;
            }
        }
        for(int ele : arr)
            System.out.print(ele+" ");
    }
}

